File: ios/viewmodel/CheckInViewModel.swift
import Foundation
import SwiftUI

@MainActor
class CheckInViewModel: ObservableObject {

    @Published var history: [String] = []
    @Published var status: String = ""
    @Published var isLoading: Bool = false
    @Published var error: String?

    private let api = SafeConnectAPI.shared
    private let store = LocalStore()
    private var userId: String?

    init() {
        history = store.loadHistory()
        userId = store.loadUserId()
    }

    func sendCheckIn(type: String) async {
        isLoading = true
        error = nil
        status = "Sending..."

        do {
            if userId == nil {
                let user = try await api.createUser(name: "iOS User")
                userId = user.id
                store.saveUserId(user.id)
            }

            let response = try await api.sendCheckIn(
                userId: userId!,
                type: type
            )

            let item = "\(type) → \(response.timestamp)"
            history.append(item)
            store.saveHistory(history)

            status = "Done!"
        } catch {
            self.error = error.localizedDescription
        }

        isLoading = false
    }

    func clearHistory() {
        history.removeAll()
        store.clearHistory()
        status = "History cleared"
    }
}

File: android/viewmodel/CheckInViewModel.kt
package com.safeconnect.app.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.safeconnect.app.network.CheckInType
import com.safeconnect.app.network.RetrofitClient
import com.safeconnect.app.repository.CheckInRepository
import com.safeconnect.app.storage.LocalStore
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch

class CheckInViewModel(
    private val localStore: LocalStore
) : ViewModel() {

    private val repository = CheckInRepository(RetrofitClient.api)

    private val _status = MutableStateFlow("")
    val status: StateFlow<String> = _status

    private val _history = MutableStateFlow<List<String>>(emptyList())
    val history: StateFlow<List<String>> = _history

    private var userId: String? = null

    init {
        // Cargar historial guardado al abrir la app
        viewModelScope.launch {
            localStore.historyFlow().collect { saved ->
                _history.value = saved
            }
        }
    }

    fun sendCheckIn(type: CheckInType) {
        viewModelScope.launch {
            _status.value = "Sending..."

            if (userId == null) {
                val user = repository.createUser("Android User")
                userId = user.id
            }

            val response = repository.sendCheckIn(userId!!, type)
            val newItem = "${type.name} → ${response.timestamp}"

            val newHistory = _history.value + newItem
            _history.value = newHistory

            // Guardar historial local
            localStore.saveHistory(newHistory)

            _status.value = "Done!"
        }
    }
}